modules/ardrivo/src/Arduino.cxx

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Arduino.cxx
+ *  Copyright 2020 ItJustWorkTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include <BoardData.hxx>
+#include <Arduino.h>
+#include <Entrypoint.hxx>
+
+using namespace std::literals;
+
+BoardData data;
```

Comment:
This global variable will be used throughout the entirety of the program; `data` is really generic and context-dependent; `board_data` may be a better fit

Reply:
While you're at it, `BoardData.hxx` should probably declare this symbol as `extern` too

Reply:
Oh and make it a pointer, because `init` will initialize it from the runtime